pallets/pool-system/src/tranches.rs

```diff
@@ -1394,12 +1577,2336 @@ where
+						Rate::zero(),
+						Rate::zero(),
+						Rate::saturating_from_rational(1, 3),
+					])
+				);
+			}
+
+			#[test]
+			/// Check price evolution over course of multiple years without adjusting total assets.
+			///
+			/// Each tranche has a different APR, debt, reserve and total issuance.
+			/// The sum of total issuance (initial NAV) for all three tranches is 1000.
+			fn calculate_prices_last_update_works() {
+				let mut tranches = default_tranches_with_issuance();
+				assert_eq!(
+					tranches.calculate_prices::<_, TTokens, TrancheCurrency>(
+						1_100_000_000,
+						SECS_PER_YEAR
+					),
+					Ok(vec![
+						Rate::saturating_from_rational(1396143445, 1000000000),
+						Rate::saturating_from_rational(1021034184, 1000000000),
+						Rate::saturating_from_rational(103418073, 100000000),
+					])
+				);
+				// increase time since last update by two to reduce res price
+				assert_eq!(
+					tranches.calculate_prices::<_, TTokens, TrancheCurrency>(
+						1_100_000_000,
+						2 * SECS_PER_YEAR
+					),
+					Ok(vec![
+						Rate::saturating_from_rational(1284127705, 1000000000),
+						Rate::saturating_from_rational(1044280552, 1000000000),
+						Rate::saturating_from_rational(
+							1070113943333333333333333333u128,
+							1000000000000000000000000000u128
+						),
+					])
+				);
+				// increase time since last update by ten to reduce res and first non-res prices
+				assert_eq!(
+					tranches.calculate_prices::<_, TTokens, TrancheCurrency>(
+						1_100_000_000,
+						5 * SECS_PER_YEAR
+					),
+					Ok(vec![
+						Rate::saturating_from_rational(89161395, 100000000),
+						Rate::saturating_from_rational(1129744254, 1000000000),
+						Rate::saturating_from_rational(
+							1189350276666666666666666667u128,
+							1000000000000000000000000000u128
+						),
+					])
+				);
+				// increase time since last update by twenty to reduce
+				assert_eq!(
+					tranches.calculate_prices::<_, TTokens, TrancheCurrency>(
+						1_100_000_000,
+						20 * SECS_PER_YEAR
+					),
+					Ok(vec![
+						Rate::zero(),
+						Rate::saturating_from_rational(91268727, 100000000),
+						Rate::saturating_from_rational(
+							2145521216666666666666666667u128,
+							1000000000000000000000000000u128
+						),
+					])
+				);
+			}
+
+			// FIXME: Are we fine with rounding errors up to 0.5%?
+			// #[test]
+			fn calculate_prices_rounding_works() {
+				let mut tranches = default_tranches_with_issuance();
+				assert_ok!(tranches.calculate_prices::<Rate, TTokens, TrancheCurrency>(
+					1_100_000_000,
+					SECS_PER_YEAR
+				));
+
+				for i in 2..20 {
+					assert_eq!(
+						tranches.calculate_prices::<Rate, TTokens, TrancheCurrency>(
+							1_100_000_000,
+							i * SECS_PER_YEAR
+						),
+						default_tranches_with_issuance()
+							.calculate_prices::<Rate, TTokens, TrancheCurrency>(
+								1_100_000_000,
+								i * SECS_PER_YEAR
+							),
+						"APR != APY after {} years",
+						i
+					);
+				}
+			}
```

Comment:
No changes with higher numbers? 

Can we try with explicit rounding through `FixedPointNumberExtension`

Reply:
The comment was off, even without rounding the error % is around `10e-6`% or lower, see a7fdfb4d0ebb981f359b57268c032fc20c48d302 where I added rounding with a max precision of `10e-6`%.

Reply:
Yes, this is fine .